x/ibc/03-connection/types/version.go

```diff
@@ -1,96 +1,45 @@
 package types
 
-// GetCompatibleVersions returns an ordered set of compatible IBC versions for the
-// caller chain's connection end.
-func GetCompatibleVersions() []string {
-	return []string{"1.0.0"}
-}
-
-// LatestVersion gets the latest version of a connection protocol
-//
-// CONTRACT: version array MUST be already sorted.
-func LatestVersion(versions []string) string {
-	if len(versions) == 0 {
-		return ""
-	}
-	return versions[len(versions)-1]
-}
-
-// PickVersion picks the counterparty latest version that is matches the list
-// of compatible versions for the connection.
-func PickVersion(counterpartyVersions, compatibleVersions []string) string {
-
-	n := len(counterpartyVersions)
-	m := len(compatibleVersions)
-
-	// aux hash maps to lookup already seen versions
-	counterpartyVerLookup := make(map[string]bool)
-	compatibleVerLookup := make(map[string]bool)
-
-	// versions suported
-	var supportedVersions []string
-
-	switch {
-	case n == 0 || m == 0:
-		return ""
-	case n == m:
-		for i := n - 1; i >= 0; i-- {
-			counterpartyVerLookup[counterpartyVersions[i]] = true
-			compatibleVerLookup[compatibleVersions[i]] = true
-
-			// check if we've seen any of the versions
-			if _, ok := compatibleVerLookup[counterpartyVersions[i]]; ok {
-				supportedVersions = append(supportedVersions, counterpartyVersions[i])
-			}
+import (
+	"strings"
 
-			if _, ok := counterpartyVerLookup[compatibleVersions[i]]; ok {
-				// TODO: check if the version is already in the array
-				supportedVersions = append(supportedVersions, compatibleVersions[i])
-			}
-		}
-	case n > m:
-		for i := n - 1; i >= m; i-- {
-			counterpartyVerLookup[compatibleVersions[i]] = true
-		}
-
-		for i := m - 1; i >= 0; i-- {
-			counterpartyVerLookup[counterpartyVersions[i]] = true
-			compatibleVerLookup[compatibleVersions[i]] = true
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
 
-			// check if we've seen any of the versions
-			if _, ok := compatibleVerLookup[counterpartyVersions[i]]; ok {
-				supportedVersions = append(supportedVersions, counterpartyVersions[i])
-			}
+var (
+	// DefaultIBCVersion represents the latest supported version of IBC
+	DefaultIBCVersion = "1.0.0"
```

Comment:
maybe we should add a validation function for versions on 24-host to match this regex and add validation on messages too 👍

Reply:
Makes sense, though I would like consensus on if we are using "1.0.0" as the 1.0 version string before adding in validation that requires semantic versioning

Reply:
I ended up applying this suggestion anyways. I figured it wouldn't take long and we could revert if needed